libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ProbabilityDistributionSplineChart.tsx

```diff
@@ -27,14 +34,86 @@ export class ProbabilityDistributionSplineChart extends React.Component<
   public context: React.ContextType<typeof ModelAssessmentContext> =
     defaultModelAssessmentContext;
 
+  public constructor(props: IProbabilityDistributionSplineChartProps) {
+    super(props);
+    this.state = {
+      probabilityBinCountArray: [],
+      selectedCohorts: this.props.selectedCohorts
+    };
+  }
+
+  public componentDidMount(): void {
+    this.calculateSplineDataIfNeeded();
+  }
+
+  public componentDidUpdate(
+    prevProps: IProbabilityDistributionSplineChartProps
+  ): void {
+    this.calculateSplineDataIfNeeded(prevProps);
+  }
+
+  public async calculateSplineDataIfNeeded(
+    prevProps?: IProbabilityDistributionSplineChartProps
+  ): Promise<void> {
+    if (
+      prevProps === undefined ||
```

Comment:
why do we need this?

Reply:
The prevProps can be undefined. The ProbabilityDistributionBoxChart.tsx follow a similar condition style.